### DISCUSSION TITLE
RADIOLIB_LORAWAN_NO_DOWNLINK disappeard from Version 6.6 to 7.00

### DISCUSSION THREAD

**Discussion by decodeais** [jgromes/RadioLib#1245] [Q&A]
If I compile the **LoRawan_TTN** example for the  **Heltec_Wifi LoRa32(v3)** the compiler misses  **RADIOLIB_LORAWAN_NO_DOWNLINK**. 
 Is this a mistake on your side, or has the example program to be changed.
For me it works with the downgraded RadioLib. [Issue on Heltec_Wifi LoRa32(v3) ](https://github.com/ropg/heltec_esp32_lora_v3/issues/51#issue-2555093826)

**Setup:**
- Board: Heltec WiFi LoRa 32 (V3)
- Library version**: LoRaWan_ESP v1.1.0
- Heltec_ESP32_LoRa v0.9.1
- RadioLib  v7.0.0
- ESP8266 and ESP32  OLED Driver ... v4.6.1

**Comment by StevenCellist** [comment]
If you take a look at the updated examples for LoRaWAN in 7.0, you will see that there have been some modifications (as mentioned in the release message). Please have a look at these examples and modify your own sketch accordingly. And yes, this also means that NO_DOWNLINK has been removed.

If you run into problems, please let us know and we'll help you. But as this is not an actual issue with the library, I'm converting it to a discussion instead.

**Reply by decodeais** [reply]
I need a example for a TTN node. The only programm for the **Heltec WiFi LoRa 32 (V3)** is the **LoRawan_TTN** . When I looked to the RadioLib examples I found nothing comparable. 
At this point it gets difficult.

**Reply by StevenCellist** [reply]
In that case, please refer to the library you are using - that person needs to update their examples to account for RadioLib 7.0. Or implement the changes yourself based on the examples provided here.

**Reply by HeadBoffin** [reply]
Or use the LoRaWAN examples supplied in RadioLib here as they are primarily tested against TTN and the primary device used for testing is a Heltec WiFi LoRa 32 (v3) - so it doesn't get any better than this! There is also a RadioLib persistence example that is, again, primarily tested using TTN & this board [here](https://github.com/radiolib-org/radiolib-persistence).

FYI, NO_DOWNLINK was removed as it made it sound like an uplink had failed if the message returned was NO_DOWNLINK - because most times there isn't going to be one!

On the plus side, we consider the LoRaWAN implementation to be out of beta - it was passed all the LoRa pre-certification tests - so there shouldn't be any breaking changes to the API.
